tests/OmniSharp.MSBuild.Tests/ProjectWithAnalyzersTests.cs

```diff
@@ -17,30 +17,57 @@ public ProjectWithAnalyzersTests(ITestOutputHelper output)
         {
         }
 
+        [Fact]
+        public async Task WhenProjectIsRestoredThenReanalyzeProject()
+        {
+            using (var testProject = await TestAssets.Instance.GetTestProjectAsync("ProjectWithAnalyzers"))
+            using (var host = CreateMSBuildTestHost(testProject.Directory, configurationData: TestHelpers.GetConfigurationDataWithAnalyzerConfig(roslynAnalyzersEnabled: true)))
+            {
+                await host.RestoreProject(testProject);
+
+                var diagnostics = await host.RequestCodeCheckAsync(Path.Combine(testProject.Directory, "Program.cs"));
+
+                Assert.NotEmpty(diagnostics.QuickFixes);
+                Assert.Contains(diagnostics.QuickFixes, x => x.ToString().Contains("IDE0060")); // Unused args.
+            }
+        }
+
         [Fact]
         public async Task WhenProjectHasAnalyzersItDoesntLockAnalyzerDlls()
         {
             using (var testProject = await TestAssets.Instance.GetTestProjectAsync("ProjectWithAnalyzers"))
             {
+                // TODO: Restore when host is running doesn't reload new analyzer references yet, move this
+                // after host start after that is fixed.
                 await RestoreProject(testProject);
 
                 using (var host = CreateMSBuildTestHost(testProject.Directory, configurationData: TestHelpers.GetConfigurationDataWithAnalyzerConfig(roslynAnalyzersEnabled: true)))
                 {
-                    var diagnostics = await host.RequestCodeCheckAsync(Path.Combine(testProject.Directory, "Program.cs"));
                     var analyzerReferences = host.Workspace.CurrentSolution.Projects.Single().AnalyzerReferences.ToList();
 
                     Assert.NotEmpty(analyzerReferences);
-                    Assert.NotEmpty(diagnostics.QuickFixes);
 
                     // This should not throw when analyzers are shadow copied.
                     Directory.Delete(Path.Combine(testProject.Directory, "./nugets"), true);
                 }
             }
         }
 
+        [Fact]
+        public async Task LoadProjectWithNewCompilationOptions()
```

Comment:
>sk LoadProjectWithNewCompilationOptions() [](start = 23, length = 41)

How is this different from the one in ProjectFileInfoTests?

Reply:
At `ProjectFileInfoTests` tested that csproj is loaded to project info. 

Since then test `LoadProjectWithNewCompilationOptions` was added which tests that rulesets get correctly loaded to project compilation options. I feel that test in `ProjectFileInfoTests` is obsolete duplicate because of that. 

Removed test from `ProjectFileInfoTests`. 

Renamed `LoadProjectWithNewCompilationOptions` -> `WhenProjectIsLoadedThenItContainsCustomRulesetsFromCsproj`.